$OpenBSD: patch-src_dvdnav_dvdnav_h,v 1.1 2012/09/26 00:04:09 brad Exp $

- Add a function to create a dpu of a dvdnav handle
- Fix C++ compatibility broken in r1208
- Allow backward search for SEEK_CUR

--- src/dvdnav/dvdnav.h.orig	Sat Feb 26 15:32:32 2011
+++ src/dvdnav/dvdnav.h	Tue Sep 11 20:35:47 2012
@@ -89,6 +89,9 @@ typedef int32_t dvdnav_status_t;
  */
 dvdnav_status_t dvdnav_open(dvdnav_t **dest, const char *path);
 
+dvdnav_status_t dvdnav_dup(dvdnav_t **dest, dvdnav_t *src);
+dvdnav_status_t dvdnav_free_dup(dvdnav_t *this);
+
 /*
  * Closes a dvdnav_t previously opened with dvdnav_open(), freeing any
  * memory associated with it.
@@ -281,7 +284,7 @@ dvdnav_status_t dvdnav_part_play(dvdnav_t *self, int32
 /*
  * Plays the specified title, starting from the specified program
  */
-dvdnav_status_t dvdnav_program_play(dvdnav_t *this, int32_t title, int32_t pgcn, int32_t pgn);
+dvdnav_status_t dvdnav_program_play(dvdnav_t *self, int32_t title, int32_t pgcn, int32_t pgn);
 
 /*
  * Stores in *times an array (that the application *must* free) of
@@ -370,7 +373,7 @@ dvdnav_status_t dvdnav_part_search(dvdnav_t *self, int
  * fcntl.h.
  */
 dvdnav_status_t dvdnav_sector_search(dvdnav_t *self,
-				     uint64_t offset, int32_t origin);
+				     int64_t offset, int32_t origin);
 
 /*
  returns the current stream time in PTS ticks as reported by the IFO structures
